dell/src/main/java/org/apache/iceberg/dell/impl/EcsAppendOutputStream.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.dell.impl;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.Headers;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import java.io.ByteArrayInputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import org.apache.iceberg.dell.ObjectKey;
+
+/**
+ * use ECS append api to write data
+ */
+public class EcsAppendOutputStream extends OutputStream {
```

Comment:
Why doesn't this implement `PositionOutputStream`?

Reply:
The `PositionOutputStream` is just a simple extension of `OutputStream`. To implement it in this class is a little wired in OOP. I'll create an adapter class in the next PR.